homework/index.js

```diff
@@ -27,21 +31,136 @@
+      const noDescription = createAndAppend('li', info, {});
+      createAndAppend('span', noDescription, {
+        text: 'Description: there is no description',
+        class: '',
+      });
+    } else {
+      const liDescription = createAndAppend('li', info, {});
+      createAndAppend('span', liDescription, { text: 'Description: ' });
+      createAndAppend('span', liDescription, { text: repositoryDescription, class: 'result' });
+    }
+
+    const link = createAndAppend('li', info, {});
+    createAndAppend('span', link, { text: 'link: ' });
+    createAndAppend('a', link, {
+      text: dataName,
+      target: '_blank',
+      href: `https://github.com/HackYourFuture/${dataName}`,
+      class: 'result',
+    });
+    const liOfForks = createAndAppend('li', info, {});
+    createAndAppend('span', liOfForks, { text: 'Forks: ' });
+    createAndAppend('span', liOfForks, { text: fork, class: 'result' });
+    const liUpdateAt = createAndAppend('li', info, {});
+    createAndAppend('span', liUpdateAt, { text: 'updated_at: ' });
+    createAndAppend('span', liUpdateAt, { text: updated, class: 'result' });
+
+    // call all contributors for this selected repository
+    fetchContributorsAndRender(contributorsUrl);
+  }
+
+  function reloadAllRepositories(parenElement, allData) {
+    const pre = document.getElementById('pre');
+    const parent = createAndAppend('div', parenElement, { class: 'bodyInformation' });
+    createAndAppend('div', parent, { id: 'info', class: 'info' });
+    createAndAppend('div', parent, { id: 'cont', class: 'cont' });
+
+    const selector = createAndAppend('select', pre, { id: 'select', class: 'select' });
+    allData.forEach((data, index) => {
+      createAndAppend('option', selector, { text: data.name, value: index });
+    });
+    // reload default repository while reloading page.
+    reloadAndRenderAllData(allData[0]);
+
+    // reload repository once selected a repository.
+    selector.addEventListener('change', function changeMe() {
+      reloadAndRenderAllData(allData[this.value]);
     });
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  // async Repository's
+  async function fetchAllRepository(url) {
+    try {
+      const data = await fetchJSON(url);
+      const root = document.getElementById('root');
+      data.sort((x, y) => x.name.localeCompare(y.name));
+      reloadAllRepositories(root, data);
+    } catch (error) {
+      const root = document.getElementById('root');
+      createAndAppend('div', root, {
+        text: "there is no repository's available",
+        class: 'alert-error',
+      });
+    }
+  }
 
+  const HYF_REPOS_URL = `https://api.github.com/orgs/HackYourFuture/repos?per_page=100`;
+
+  function main(url) {
+    const root = document.getElementById('root');
+    const pre = createAndAppend('pre', root, { class: 'pre', id: 'pre' });
+    createAndAppend('span', pre, { text: 'HYF Repositories', class: 'logoName' });
```

Comment:
Ah. The `class` attribute value should preferably be in `kebab-case` rather than `camelCase` i.e it should be `logo-name` instead of `logoName` 🙂

Reply:
Yes, you are right here, thank you Yash 🙂